handler.go

```diff
@@ -349,12 +357,24 @@ func (s *handler) createError(err error) *respError {
 	}
 
 	if m, ok := err.(marshalable); ok {
-		meta, err := m.MarshalJSON()
-		if err == nil {
+		meta, marshalErr := m.MarshalJSON()
+		if marshalErr == nil {
 			out.Meta = meta
+		} else {
+			log.Warnf("Failed to marshal error metadata: %v", marshalErr)
 		}
 	}
 
+	var err2 Error
+	if errors.As(err, &err2) {
+		out.Code = ErrorCode(err2.ErrorCode())
```

Comment:
Why are we setting the code here again ? Don't we already have it on line 355 above ?

Reply:
@aarshkshah1992 
So in line 355 we are setting the registered error codes, but here we are setting the custom error code.

I think we can directly register the `code` for `ErrWithExecutionReverted`, but then we need to finalise on which code to use, as error code `3` is already registered with the `ActorNotFound`